File: Modules/CustomerManagement/CustomerContact/CustomerContactColumns.cs
namespace MapPin.CustomerManagement.Forms;

[ColumnsScript("CustomerManagement.CustomerContact")]
[BasedOnRow(typeof(CustomerContactRow), CheckNames = true)]
public class CustomerContactColumns
{
    [EditLink, Width(180)]
    public string ContactName { get; set; }

    [Width(130)]
    public string Title { get; set; }

    [Width(120)]
    public string Phone { get; set; }

    [Width(160)]
    public string Email { get; set; }

    public string Notes { get; set; }
}

File: Modules/ServerTypes/CustomerManagement/CustomerContactColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { CustomerContactRow } from "./CustomerContactRow";

export interface CustomerContactColumns {
    ContactName: Column<CustomerContactRow>;
    Title: Column<CustomerContactRow>;
    Phone: Column<CustomerContactRow>;
    Email: Column<CustomerContactRow>;
    Notes: Column<CustomerContactRow>;
}

export class CustomerContactColumns extends ColumnsBase<CustomerContactRow> {
    static readonly columnsKey = 'CustomerManagement.CustomerContact';
    static readonly Fields = fieldsProxy<CustomerContactColumns>();
}